packages/renderer/src/lib/table/Table.svelte

```diff
@@ -0,0 +1,165 @@
+<style>
+.grid-table {
+  display: grid;
+}
+</style>
+
+<script lang="ts">
+/* eslint-disable import/no-duplicates */
+// https://github.com/import-js/eslint-plugin-import/issues/1479
+import { afterUpdate, tick } from 'svelte';
+import Checkbox from '../ui/Checkbox.svelte';
+import type { Column, Row } from './table';
+import { flip } from 'svelte/animate';
+/* eslint-enable import/no-duplicates */
+
+export let kind: string;
+export let data: any[];
+export let columns: Column<any>[];
+export let row: Row<any>;
+
+// number of selected items in the list
+export let selectedItemsNumber: number = 0;
+$: selectedItemsNumber = row.selectable
+  ? data.filter(object => row.selectable?.(object)).filter(object => object.selected).length
+  : 0;
+
+// do we need to unselect all checkboxes if we don't have all items being selected ?
+$: selectedAllCheckboxes = row.selectable
+  ? data.filter(object => row.selectable?.(object)).every(object => object.selected)
+  : false;
+
+function toggleAll(checked: boolean) {
+  if (!row.selectable) {
+    return;
+  }
+  const toggleData = data;
+  toggleData.filter(object => row.selectable?.(object)).forEach(object => (object.selected = checked));
+  data = toggleData;
```

Comment:
What is the purpose of using the `toggleData` temp variable? As toggleData is a reference to data, it does not seem necessary to me

Reply:
You can see this was actually unchanged from before. Svelte only detects changed objects, not internal state, so sometimes you have to explicitly assign something to see the change visually. I did some testing and this doesn't seem to be required anymore, so I removed it.

The same thing existed further down on line 64 with sorting. This one is required, but I can see the same question arising so I changed it to an explicit 'data = data.sort()', and added the required no-mutation comment. I don't think this is any better code-wise, but at least it's obvious it was intentional.

Reply:
Now that you tell me the reason, I remember I effectively had to write some `data = [...data]` in some other PR, to make the refresh happen. Not sure what is the recommended way to handle this

Reply:
In this article, it is explained that `data = data` is enough to trigger the Svelte changes.

https://javascript.plainenglish.io/update-arrays-and-objects-in-svelte-40c6a84f8a28

Reply:
I've seen it elsewhere too so it seem to be a common way of handling it. For now though sort is the only place that needs it and eslint comment make it clear the code is intentional.